ShellCheck/Parser.hs

```diff
@@ -150,13 +182,15 @@ data HereDocContext =
 data UserState = UserState {
     lastId :: Id,
     positionMap :: Map.Map Id SourcePos,
```

Comment:
maybe this should be `Map Id (SourcePos, Maybe SourcePos)` with a goal of removing the Maybe at some later point, rather than adding a separate map?